ritasamir/assignment1/Q_02.java

```diff
@@ -0,0 +1,11 @@
+package assignment_1;
+
+import java.util.LinkedList;
+
+public class Q_02<T> {
+
+	public T getLastK(LinkedList<T> linkedList, int k) {
+		int position = linkedList.size() - k - 1;

```

Comment:
Very smart. :) This would be a good solution in a real world situation, but here for the sake of the assignment I would like to suggest we use a slightly different interface for the list.
The difficulty in this assignment is that in a singly linked list we don't know the size upfront. We only know the start element, the end element and each element of the list has a pointer/reference to the next one (the last one has a null pointer/reference).
Do you think you could give it another try using an Iterator? If you don't like iterators, I can provide an interface and implementation of a singly linked list for you to work with.